eis_toolkit/vector_processing/kriging_interpolation.py

```diff
@@ -0,0 +1,106 @@
+from numbers import Number
+
+import geopandas as gpd
+import numpy as np
+import numpy.ma as ma
+from beartype import beartype
+from beartype.typing import Tuple
+from pykrige.ok import OrdinaryKriging
+from pykrige.uk import UniversalKriging
+
+from eis_toolkit.exceptions import (
+    EmptyDataFrameException,
+    InvalidParameterValueException,
+    NotApplicableGeometryTypeException,
+)
+
+
+def _kriging(
+    data: gpd.GeoDataFrame,
+    resolution: Tuple[Number, Number],
+    extent: Tuple[Number, Number, Number, Number],
+    variogram_model: str,
+    method: str,
+    drift_terms: list,
+) -> Tuple[np.ndarray, dict]:
+
+    coordinates = np.array(list(data.geometry.apply(lambda geom: [geom.x, geom.y, geom.z])))
+    x = coordinates[:, 0]
+    y = coordinates[:, 1]
+    z = coordinates[:, 2]
+
+    grid_x = np.linspace(extent[0], extent[1], resolution[0])
+    grid_y = np.linspace(extent[2], extent[3], resolution[1])
+
+    if method == "universal":
+        universal_kriging = UniversalKriging(x, y, z, variogram_model=variogram_model, drift_terms=drift_terms)
+        z_interpolated, _ = universal_kriging.execute("grid", grid_x, grid_y)
+
+    if method == "ordinary":
+        ordinary_kriging = OrdinaryKriging(x, y, z, variogram_model=variogram_model)
+        z_interpolated, _ = ordinary_kriging.execute("grid", grid_x, grid_y)
+
+    z_interpolated = ma.getdata(z_interpolated)
+
+    out_meta = {"crs": data.crs, "width": len(grid_x), "height": len(grid_y)}
+
+    return z_interpolated, out_meta
+
+
+@beartype
+def kriging(
+    data: gpd.GeoDataFrame,
+    resolution: Tuple[Number, Number],
+    extent: Tuple[Number, Number, Number, Number],
+    variogram_model: str = "linear",
+    method: str = "ordinary",
```

Comment:
These 2 parameters could be changed to use Literals. This way you could get rid of the separate checks below (beartype would check these instead). You can still test these parameter violations in tests, just change the expected exception to beartype's exception there.
```suggestion
    variogram_model: Literal["linear", "power", "gaussian", "spherical", "exponential", "hole-effect"] = "linear",
    method: Literal["ordinary", "universal"] = "ordinary",
```